pkg/cmd/runbook/snapshot/publish/publish.go

```diff
@@ -0,0 +1,238 @@
+package publish
+
+import (
+	"errors"
+	"fmt"
+	"github.com/MakeNowJust/heredoc/v2"
+	"github.com/OctopusDeploy/cli/pkg/cmd"
+	"github.com/OctopusDeploy/cli/pkg/cmd/runbook/shared"
+	"github.com/OctopusDeploy/cli/pkg/constants"
+	"github.com/OctopusDeploy/cli/pkg/factory"
+	"github.com/OctopusDeploy/cli/pkg/output"
+	"github.com/OctopusDeploy/cli/pkg/question/selectors"
+	"github.com/OctopusDeploy/cli/pkg/util/flag"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/projects"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/runbooks"
+	"github.com/spf13/cobra"
+	"math"
+	"time"
+)
+
+const (
+	FlagProject  = "project"
+	FlagRunbook  = "runbook"
+	FlagSnapshot = "snapshot"
+)
+
+type PublishFlags struct {
+	Runbook  *flag.Flag[string]
+	Project  *flag.Flag[string]
+	Snapshot *flag.Flag[string]
+}
+
+func NewPublishFlags() *PublishFlags {
+	return &PublishFlags{
+		Runbook:  flag.New[string](FlagRunbook, false),
+		Project:  flag.New[string](FlagProject, false),
+		Snapshot: flag.New[string](FlagSnapshot, false),
+	}
+}
+
+type PublishOptions struct {
+	*PublishFlags
+	*shared.RunbooksOptions
+	GetAllProjectsCallback shared.GetAllProjectsCallback
+	*cmd.Dependencies
+}
+
+func NewPublishOptions(publishFlags *PublishFlags, dependencies *cmd.Dependencies) *PublishOptions {
+	return &PublishOptions{
+		PublishFlags:           publishFlags,
+		RunbooksOptions:        shared.NewGetRunbooksOptions(dependencies),
+		GetAllProjectsCallback: func() ([]*projects.Project, error) { return shared.GetAllProjects(dependencies.Client) },
+		Dependencies:           dependencies,
+	}
+}
+
+func NewCmdPublish(f factory.Factory) *cobra.Command {
+	publishFlags := NewPublishFlags()
+	cmd := &cobra.Command{
+		Use:     "publish",
+		Short:   "Publish a runbook snapshot",
+		Long:    "Publish a runbook snapshot in Octopus Deploy",
+		Aliases: []string{"new", "publish"},
```

Comment:
do we need to alias the main command name? 😄 
```suggestion
		Aliases: []string{"new"},
```

Reply:
Actually, on second thought...wouldn't `new` here clash with the `new` alias for the `create` command?

Reply:
I think this might be a copy paste mistake, I don't think we need an alias here

Reply:
The aliases have been removed